network/dhcpd.py

```diff
@@ -0,0 +1,181 @@
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: dhcpd
+short_description: Simple module that sets/reads entries from an ISC-DHCP server
+description:
+    - Module that interfaces with an ISC-DHCP server using OMAPI to add/remove host objects or lookup lease objects
+version_added: "0.0"
+author: Vyronas Tsingaras
+requirements:
+    - This module requires the Python pypureomapi module
+    - You should have configured your dhcp server to enable OMAPI
+options:
+    option_name:
+        state:
+            - If state is ommited then lookup a lease and return as a fact, else perform the respective operation
+        required: false
+        default: null
+        choices: [present, absent]
+        version_added: 1.0
+        mac_address:
+            - Target MAC address, if ommited ip_address must be supplied
+        required: false
+        default: null
+        version_added: 1.0
+        ip_address:
+            - Target IPv4 address, if ommited mac_address must be supplied
+        required: false
+        default: null
+        version_added: 1.0
+        server:
+            - IP or hostname of DHCP server
+        required: true
+        default: null
+        version_added: 1.0
+        port:
+            - OMAPI DHCP port
+        required: false
+        default: 7911
+        version_added: 1.0
+        key_name:
+            - OMAPI key name as specified in the omapi-key directive in dhcpd.conf
+        required: false
+        default: null
+        version_added: 1.0
+        key_secret:
+            - base64-encoded OMAPI key secret as specified in the "key $KEYNAME" stanza in dhcpd.conf
+        required: false
+        default: null
+        version_added: 1.0
+'''
+
+EXAMPLES = '''
+# Add a new host object entry in dhcpd.leases
+- dhcpd: state=present ip_address=10.0.0.10 mac_address=13:37:be:ef:00:00 server=dhcp.example.org key_name=omapi_key key_secret="base64encodedsecretgoeshere"
+# Remove a host object entry in dhcpd.leases by its mac address
+- dhcpd: state=absent mac_address=13:37:be:ef:00:00 server=dhcp.example.org
+# Lookup a lease by its IP address, its ip and mac addresses will be registered as dhcp_mac and dhcp_ip
+- dhcpd: ip_address=10.0.0.10 server=dhcp.example.org
+'''
+
+import json
+
+try:
+    import pypureomapi
+    HAVE_PYPUREOMAPI=True
+except ImportError:
+    HAVE_PYPUREOMAPI=False
+
+def main():
+    module = AnsibleModule(
+        argument_spec = dict(
+            state   = dict(default=None, choices=['present', 'absent']),
```

Comment:
What if I add a new gather_facts option and stop relying on state? 

Brian Coca <notifications@github.com> wrote:

In network/dhcpd.py:

> +# Lookup a lease by its IP address, its ip and mac addresses will be registered as dhcp_mac and dhcp_ip > +- dhcpd: ip_address=10.0.0.10 server=dhcp.example.org > +''' > + > +import json > + > +try: > + import pypureomapi > + HAVE_PYPUREOMAPI=True > +except ImportError: > + HAVE_PYPUREOMAPI=False > + > +def main(): > + module = AnsibleModule( > + argument_spec = dict( > + state = dict(default=None, choices=['present', 'absent']), 

i would split that into a dhcp_facts module instead of abusing the lack of state

—
Reply to this email directly or view it on GitHub. 



Sent with mySecureMail.
http://www.mysecurephone.eu/